app/views/transactions/_pagination.html.erb

```diff
@@ -1,39 +1,48 @@
-        <%= link_to pagy_url_for(pagy, series_item), class: "inline-flex items-center px-3 py-3 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700" do %>
-          <%= series_item %>
+    </div>
+    <div class="bg-gray-25 rounded-xl">
+      <% pagy.series.each do |series_item| %>
+        <% if series_item.is_a?(Integer) %>
+          <%= link_to pagy_url_for(pagy, series_item), class: "inline-flex items-center px-3 py-3 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700" do %>
+            <%= series_item %>
+          <% end %>
+        <% elsif series_item.is_a?(String) %>
+          <%= link_to pagy_url_for(pagy, series_item), class: "shadow-lg ring-2 ring-inset ring-gray-200 rounded-xl bg-white inline-flex items-center m-1 px-4 py-2 text-sm font-medium text-gray-900" do %>
+            <%= series_item %>
+          <% end %>
+        <% elsif series_item == :gap %>
+          <span class="inline-flex items-center  px-3 py-3 text-sm font-medium text-gray-500">...</span>
         <% end %>
-      <% elsif series_item.is_a?(String) %>
-        <%= link_to pagy_url_for(pagy, series_item), class: "shadow-lg ring-2 ring-inset ring-gray-200 rounded-xl bg-white inline-flex items-center m-1 px-4 py-2 text-sm font-medium text-gray-900" do %>
-          <%= series_item %>
+      <% end %>
+    </div>
+    <div>
+      <% if pagy.next %>
+        <%= link_to pagy_url_for(pagy, pagy.next), class: "inline-flex items-center px-3 py-3 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700" do %>
+          <%= lucide_icon("chevron-right", class: "w-5 h-5 text-gray-500") %>
         <% end %>
-      <% elsif series_item == :gap %>
-        <span class="inline-flex items-center  px-3 py-3 text-sm font-medium text-gray-500">...</span>
+      <% else %>
+        <div class="inline-flex items-center px-3 py-3 text-sm font-medium hover:border-gray-300">
+          <%= lucide_icon("chevron-right", class: "w-5 h-5 text-gray-200") %>
+        </div>
       <% end %>
-    <% end %>
+    </div>
   </div>
-  <div>
-    <% if pagy.next %>
-      <%= link_to pagy_url_for(pagy, pagy.next), class: "inline-flex items-center px-3 py-3 text-sm font-medium text-gray-500 hover:border-gray-300 hover:text-gray-700" do %>
-        <%= lucide_icon("chevron-right", class: "w-5 h-5 text-gray-500") %>
-      <% end %>
-    <% else %>
-      <div class="inline-flex items-center px-3 py-3 text-sm font-medium hover:border-gray-300">
-        <%= lucide_icon("chevron-right", class: "w-5 h-5 text-gray-200") %>
-      </div>
-    <% end %>
+  <% per_page = params[:per_page] || "50" %>
+  <div class="flex items-center gap-4">
+  <p class="text-sm text-gray-500">Rows per page</p>
+  <%= form_with url: transactions_path, method: :get, class: "flex items-center gap-4", data: { controller: "auto-submit-form" } do %>
+    <%= render partial: "pagination_per_page_selector", locals: { value: per_page } %>
+  <% end %>
```

Comment:
As a result of the other changes that I mentioned, the custom select here will have "floating" issues where it won't be able to break out of the bounding div.

In the future, we'll be building out a design system for components like custom selects to ensure that these behaviors are handled appropriately, but right now, we've moved towards the path of using native HTML form elements until we've got this design system in place.

No way to know that on your end so no worries at all.  Here's a tested and completed replacement form with native controls that you can drop in here to get it all working:

```erb
  <%= form_with url: transactions_path,
                builder: ActionView::Helpers::FormBuilder,
                method: :get,
                class: "flex items-center gap-4",
                data: { controller: "auto-submit-form" } do |f| %>
    <%= f.label :per_page, t(".rows_per_page"), class: "text-sm text-gray-500" %>
    <%= f.select :per_page,
                 options_for_select(["10", "20", "30", "50"], params[:per_page]),
                 {},
                 class: "py-1.5 pr-8 text-sm text-gray-900 font-medium border border-gray-200 rounded-lg focus:border-gray-900 focus:ring-gray-900 focus-visible:ring-gray-900",
                 data: { "auto-submit-form-target": "auto" } %>
  <% end %>
```

Reply:
Just an FYI, you'll need to add the `.rows_per_page` key to the `config/locales/views/transactions/en.yml` locale file as "Rows per page"

Reply:
Replace custom select with browser select and move the text to localization file